opamp/observiq/observiq_client.go

```diff
@@ -229,6 +238,91 @@ func (c *Client) onRemoteConfigHandler(ctx context.Context, remoteConfig *protob
 	return nil
 }
 
+func (c *Client) onPackagesAvailableHandler(packagesAvailable *protobufs.PackagesAvailable) error {
+	c.logger.Debug("Packages available handler")
+
+	// Initialize PackageStatuses that will eventually be sent back to server
+	curPackageStatuses := &protobufs.PackageStatuses{
+		ServerProvidedAllPackagesHash: packagesAvailable.GetAllPackagesHash(),
+		Packages:                      map[string]*protobufs.PackageStatus{},
+	}
+
+	// Retrieve last known status (this should return with minimal info even on first time)
+	lastPackageStatuses, err := c.packagesStateProvider.LastReportedStatuses()
+
+	// If there is a problem retrieving the last saved PackageStatuses, we will log the error
+	// but continue on as the only thing missing will be the agent package hash.
+	if err != nil {
+		c.logger.Debug("Failed to retrieve last reported package statuses", zap.Error(err))
+	}
+
+	var lastPkgStatusMap map[string]*protobufs.PackageStatus
+	if lastPackageStatuses != nil {
+		lastPkgStatusMap = lastPackageStatuses.GetPackages()
+	}
+
+	curPackages := c.createPackageStatusMap(packagesAvailable.GetPackages(), lastPkgStatusMap)
+	curPackageStatuses.Packages = curPackages
+
+	if err = c.packagesStateProvider.SetLastReportedStatuses(curPackageStatuses); err != nil {
+		c.logger.Debug("Failed to save last reported package statuses", zap.Error(err))
```

Comment:
Should this return an error? Also, this should probably be a warning log as it would be good to know if this is happening.